app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -469,7 +469,13 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 ImageCaptureProcessing.processCaptureResponse(this, getInstanceFolder(), true);
                 break;
             case SIGNATURE_CAPTURE:
-                ImageCaptureProcessing.processCaptureResponse(this, getInstanceFolder(), false);
+                boolean saved = ImageCaptureProcessing.processCaptureResponse(this, getInstanceFolder(), false);
+                if (saved) {
+                    ImageButton nextButton = (ImageButton)this.findViewById(R.id.nav_btn_next);
+                    if (nextButton.getTag().equals(NAV_STATE_NEXT) && !questionsView.isQuestionList()) {
```

Comment:
Worth noting but nothing block: this will run into this issue http://manage.dimagi.com/default.asp?186991